android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/temporary/share/ShareDownloadFeature.kt

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.downloads.temporary.share
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeout
+import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.action.ShareInternetResourceAction
+import mozilla.components.browser.state.selector.findTabOrCustomTabOrSelectedTab
+import mozilla.components.browser.state.state.content.ShareInternetResourceState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.fetch.Client
+import mozilla.components.feature.downloads.temporary.TemporaryDownloadFeature
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.ktx.android.content.shareMedia
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+
+/**
+ * [Middleware] implementation for sharing online resources.
+ *
+ * This will intercept only [ShareInternetResourceAction] [BrowserAction]s.
+ *
```

Comment:
Saw that these lines exist in the current implementation (thinking that they remained from a previous implementation based on middlewares).
They are confusing and should be removed.
This is not a `Middleware` but a `LifecycleAwareFeature` which does not intercept actions but rather reacts to specific store updates.

Reply:
Agreed. I had originally updated these to [LifecycleAwareFeature] but thought maybe they were kept for a reason - thanks for highlighting